internal/kafka/internal/clusters/ocm_provider_test.go

```diff
@@ -1622,3 +1622,140 @@ func TestOCMProvider_GetClusterResourceQuotaCosts(t *testing.T) {
 		})
 	}
 }
+
+func TestOCMProvider_CheckIfOrganizationIsTheClusterOwner(t *testing.T) {
+	g := gomega.NewWithT(t)
+	type fields struct {
+		ocmClient ocm.Client
+	}
+	type args struct {
+		externalOrganizationID string
+		clusterID              string
+		clusterExternalID      string
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "return an error when getting the ams organization id fails",
+			fields: fields{
+				ocmClient: &ocm.ClientMock{
+					GetOrganisationIdFromExternalIdFunc: func(externalID string) (string, error) {
+						g.Expect(externalID).To(gomega.Equal("org-id"))
+						return "", fmt.Errorf("error")
+					},
+				},
+			},
+			args: args{
+				externalOrganizationID: "org-id",
+				clusterID:              "cluster-id",
+				clusterExternalID:      "cluster-external-id",
+			},
+			wantErr: true,
+		},
+		{
+			name: "return an error when getting the organization is not found from AMS",
+			fields: fields{
+				ocmClient: &ocm.ClientMock{
+					GetOrganisationIdFromExternalIdFunc: func(externalID string) (string, error) {
+						g.Expect(externalID).To(gomega.Equal("org-id"))
+						return "", nil
+					},
+				},
+			},
+			args: args{
+				externalOrganizationID: "org-id",
+				clusterID:              "cluster-id",
+				clusterExternalID:      "cluster-external-id",
+			},
+			wantErr: true,
+		},
+		{
+			name: "return an error when getting the cluster subscription fails",
+			fields: fields{
+				ocmClient: &ocm.ClientMock{
+					GetOrganisationIdFromExternalIdFunc: func(externalID string) (string, error) {
+						g.Expect(externalID).To(gomega.Equal("org-id"))
```

Comment:
This can be done out of the mocked function:
```go
g.Expect(testcase.fields.ocmClient.GetOrganisationIdFromExternalIdCalls()[0].ExternalId).To(gomega.Equal("org-id"))
```

The only requirement is that `ocmClient` in the `fields` struct is declared as `*ocm.ClientMock`:
```go
	type fields struct {
		ocmClient *ocm.ClientMock
	}
```

Reply:
> The reason why I prefer doing it inline is that it is scoped to a testcase and avoid the need to perform conditional checks in some scenario where a particular testcase shouldn't perform this call.

Make sense